cpp/ql/test/query-tests/jsf/4.13 Functions/AV Rule 114/test.c

```diff
@@ -110,3 +110,27 @@ int f17() {
     if (__builtin_expect(1, 0))
         __builtin_unreachable(); // GOOD
 }
+
+[[_Noreturn]] void f18();
+
+int f19() {
+    f18(); // GOOD
+}
+
+[[___Noreturn__]] void f20();
```

Comment:
What's going on with `__Noreturn__` (in the change note), `___Noreturn__` (here, with the extra `_`), and the predicate itself (which list neither)?

Reply:
It should be with just two `_`, well spotted. Let me temporarily put this back in draft and fix this.

Reply:
I'm wrong. It should be three `_` as a prefix and two as a suffix, so it's just the change note that's incorrect. See: https://en.cppreference.com/w/c/language/attributes/noreturn. Note that the lower case version just has two `_` both as prefix and as suffix.

Reply:
The predicate itself does not list `___Noreturn__`, as the frontend strips off two of the `_` in the prefix and also strips off the `__` suffix. So, those do not end up in the database.

Reply:
Wow, they've made this unnecessarily fiddly!